File: LINKED_LIST/linked_list.c
#include<stdio.h>
#include<stdlib.h>

struct node
{
  int data;
  struct node *next;


};

struct node *create_node(struct node ** start, int data)
{
   struct node *newnode=(struct node *)malloc(sizeof(struct node));
   newnode->data=data;
   newnode->next=(*start);
   (*start)=newnode;


}



void printList(struct node *head)
{
  if(head==NULL)
       printf("list is empty \n");
  
  while(head!=NULL)
  {
     printf("%d  ",head->data);
     head=head->next;
  }

}



void main()
{
struct node *start=NULL;

create_node(&start,5);
create_node(&start,8);
create_node(&start,1);
create_node(&start,4);
create_node(&start,11);
create_node(&start,9);


printList(start);

}

File: LINKED_LIST/JAVA/LInkedList.java
import java.io.*;
import java.util.*;

class Node
{
   int data;
   Node next;
   Node(int d)
   {
     data=d;
     next=null;

   }
}

public class LinkedList
{
     Node head;
     public void createNode(int new_data)
     {
      Node new_node=new Node(new_data);
      new_node.next=head;
      head=new_node;
     

  public int getCount()
  {
     int count=0;
     Node temp=head;
  while(temp.next!=null)
  {
     count++;
     temp=temp.next;
  }

return count;
 }

  public void printList()
  {
     Node q=head;
     while (q.next)
     {
       System.out.print(q.data+" ");
       q=q.next;

     }

 }

 public static void main(String args[])
 {
   LinkedList list= new LinkedList();

   list.createNode(5);
   list.createNode(9);
   list.createNode(3);
   list.createNode(1);
   list.createNode(7);
   System.out.println("list :");
   list.printList();
   System.out.print("count= ",getCount());

 }

}
